src/test/java/pl/techlab24/OSKManager/model/validation/ClientValidatorTest.java

```diff
@@ -0,0 +1,243 @@
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("setOfCitiesAndValidationResults")
+    void shouldValidateCity(String city, List<String> expected) {
+        Client clientWithVariableCity = correctClient.toBuilder().city(city).build();
+
+        List<String> resultOfValidation = ClientValidator.validate(clientWithVariableCity);
+
+        assertEquals(expected, resultOfValidation);
+    }
+
+    private static Stream<Arguments> setOfCitiesAndValidationResults() {
+        return Stream.of(
+            Arguments.of(null, Collections.singletonList("Name of city cannot be null.")),
+            Arguments.of("", Collections.singletonList("Name of city must contain at least 1 character.")),
+            Arguments.of("   ", Collections.singletonList("Name of city must contain at least 1 character.")),
+            Arguments.of("Warsaw", Collections.emptyList()),
+            Arguments.of("A", Collections.emptyList()),
+            Arguments.of("Gdansk", Collections.emptyList())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("setOfPeselsAndValidationResults")
+    void shouldValidatePesel(String pesel, List<String> expected) {
+        Client clientWithVariablePesel = correctClient.toBuilder().pesel(pesel).build();
+
+        List<String> resultOfValidation = ClientValidator.validate(clientWithVariablePesel);
+
+        assertEquals(expected, resultOfValidation);
+    }
+
+    private static Stream<Arguments> setOfPeselsAndValidationResults() {
+        return Stream.of(
+            Arguments.of(null, Collections.singletonList("Pesel number cannot be null.")),
+            Arguments.of("", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("   ", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("72011517224*", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("72011517224-", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("X72011517224", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("11 12 2000", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("12/12/2012", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("12.13.2000", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("40.01.2000", Collections.singletonList("Pesel number does not match correct pesel pattern.")),
+            Arguments.of("62011517224", Collections.singletonList("Pesel number is incorrect.")),
+            Arguments.of("70020509530", Collections.singletonList("Pesel number is incorrect.")),
+            Arguments.of("72011517224", Collections.emptyList()),
```

Comment:
I would add here a pesel number that ends with 0, in order to check “if” branch from “if-else” construction from checkSum method (from ClientValidator class). I would also add a wrong pesel example number that should end with 0 but ends with another digit. Examples: 
_Arguments.of("82011617221", Collections.singletonList("Pesel number is incorrect.")),
Arguments.of("82011617220", Collections.emptyList()),_

Reply:
```suggestion
            Arguments.of("70020509530", Collections.singletonList("Pesel number is incorrect.")),
            Arguments.of("82011617221", Collections.singletonList("Pesel number is incorrect.")),
            Arguments.of("82011617220", Collections.emptyList()),
            Arguments.of("72011517224", Collections.emptyList()),
```